src/bookmarks/background/index.js

```diff
@@ -1,61 +1,20 @@
-import { dataURLToBlob } from 'blob-util'
-
-import fetchPageData from 'src/page-analysis/background/fetch-page-data'
-import * as index from 'src/search'
-import db from 'src/pouchdb'
-import { transformToBookmarkDoc } from 'src/imports'
-import { generatePageDocId } from 'src/page-storage'
 import removeBookmarkByUrl from './deletion'
+import { createBookmarkByExtension, createNewPageForBookmark } from './addition'
+import { makeRemotelyCallable } from 'src/util/webextensionRPC'
 
-async function getAttachments(pageData) {
-    const favIconBlob = await dataURLToBlob(pageData.favIconURI)
-
-    return {
-        'fav-icon-blob': {
-            content_type: favIconBlob.type,
-            data: favIconBlob,
-        },
-    }
-}
-
-async function createNewPageForBookmark(id, bookmarkInfo) {
-    let pageDoc = {
-        _id: generatePageDocId({ url: bookmarkInfo.url }),
-        url: bookmarkInfo.url,
-        title: bookmarkInfo.title,
-    }
-
-    try {
-        const pageData = await fetchPageData({
-            url: bookmarkInfo.url,
-            opts: {
-                includePageContent: true,
-                includeFavIcon: true,
-            },
-        })
-
-        pageDoc = {
-            ...pageDoc,
-            _attachments: await getAttachments(pageData),
-            content: pageData.content,
-        }
-    } catch (err) {
-        console.error(
-            'Error occurred while fetching page data: ',
-            err.toString(),
-        )
-    } finally {
-        const bookmarkDoc = transformToBookmarkDoc(pageDoc)(bookmarkInfo)
-        index.addPageConcurrent({ pageDoc, bookmarkDocs: [bookmarkDoc] })
-        db.bulkDocs([bookmarkDoc, pageDoc])
-    }
-}
+makeRemotelyCallable({ createBookmarkByExtension })
+makeRemotelyCallable({ removeBookmarkByUrl })
 
 const removeBookmarkHandler = (id, { node }) =>
     node.url
         ? removeBookmarkByUrl(node.url)
         : console.warn('Cannot remove bookmark with no URL', node)
 
+const AddBookmarkHandler = (id, node) =>
+    id
+        ? createNewPageForBookmark(id, node)
+        : createBookmarkByExtension(node.url)
```

Comment:
I don't think this condition is needed; `AddBookmarkHandler` will only ever be triggered by the `bookmarks.onCreated` event. You could just set ` createNewPageForBookmark` as the listener as it was before (I think?)